common/web/gesture-recognizer/src/engine/configuration/gestureRecognizerConfiguration.ts

```diff
@@ -81,4 +84,34 @@ export interface GestureRecognizerConfiguration<HoveredItemType> {
    * @returns
    */
   readonly itemIdentifier?: (coord: Omit<InputSample<any>, 'item'>, target: EventTarget) => HoveredItemType;
+}
+
+export function preprocessRecognizerConfig<HoveredItemType>(
+  config: GestureRecognizerConfiguration<HoveredItemType>
+): Nonoptional<GestureRecognizerConfiguration<HoveredItemType>> {
+  // Allows configuration pre-processing during this method.
+  let processingConfig: Mutable<Nonoptional<GestureRecognizerConfiguration<HoveredItemType>>> = {...config} as Nonoptional<GestureRecognizerConfiguration<HoveredItemType>>;
```

Comment:
That is one heck of a type -- starting to feel downright Java!

Reply:
I always have loved generics a little _too_ much.

Obviously not _that_ extreme, but it's been noted that [TypeScript's type system is Turing-complete](https://github.com/microsoft/TypeScript/issues/14833).

Okay, maybe I don't love them _that_ much - some people clearly had fun writing their analyses in that issue.  (Like the guy who wrote a MUD with them!)